### ISSUE TITLE
Upgrade script hangs

### ISSUE THREAD

**Issue by stefanfoulis** [thingsboard/thingsboard#2263]
I have a small installation using the `thingsboard/tb-postgres` docker image.

I pulled the latest docker image and ran `upgrade-tb.sh`. Schema upgrades seem to have worked. But it looks like the script is stuck at `Updating system data...`:

```
docker-compose run --rm tb upgrade-tb.sh
pg_ctl: another server might be running; trying to start server anyway
waiting for server to start.... done
server started
Starting ThingsBoard upgrade ...
 ===================================================
 :: ThingsBoard ::       (v2.4.2)
 ===================================================

Starting ThingsBoard Upgrade from version 2.3.1 ...
Upgrading ThingsBoard from version 2.3.1 to 2.4.0 ...
Updating schema ...
Schema updated.
Upgrading ThingsBoard from version 2.4.0 to 2.4.1 ...
Upgrading ThingsBoard from version 2.4.1 to 2.4.2 ...
Updating schema ...
Schema updated.
Updating system data...
```

There is a postgres process using 100% of one CPU core. Other than that the machine seems idle.:

```
postgres: postgres thingsboard 127.0.0.1(46408) SELECT
```

With this upgrade it has been running for `1h`. With a previous upgrade I left it running for over a day and ended up aborting the script.

How can I provide more information to better debug this?

**Comment by stefanfoulis** [comment]
I pressed `ctl+c` after `1h` and got this stacktrace... maybe it helps find out what it was doing:

```
Updating system data...
^CUnexpected error during ThingsBoard installation!
org.springframework.beans.factory.BeanCreationNotAllowedException: Error creating bean with name 'entityManagerFactory': Singleton bean creation not allowed while singletons of this factory are in destruction (Do not request a bean from a BeanFactory in a destroy method implementation!)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansOfType(DefaultListableBeanFactory.java:602)
	at org.springframework.beans.factory.BeanFactoryUtils.beansOfTypeIncludingAncestors(BeanFactoryUtils.java:340)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.detectPersistenceExceptionTranslators(PersistenceExceptionTranslationInterceptor.java:168)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:150)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:135)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:93)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.data.repository.core.support.SurroundingTransactionDetectorMethodInterceptor.invoke(SurroundingTransactionDetectorMethodInterceptor.java:61)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212)
	at com.sun.proxy.$Proxy122.findByTenantIdAndBundleAlias(Unknown Source)
	at org.thingsboard.server.dao.sql.widget.JpaWidgetTypeDao.findWidgetTypesByTenantIdAndBundleAlias(JpaWidgetTypeDao.java:54)
	at org.thingsboard.server.dao.sql.widget.JpaWidgetTypeDao$$FastClassBySpringCGLIB$$7702eb73.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:684)
	at org.thingsboard.server.dao.sql.widget.JpaWidgetTypeDao$$EnhancerBySpringCGLIB$$776bf973.findWidgetTypesByTenantIdAndBundleAlias(<generated>)
	at org.thingsboard.server.dao.widget.WidgetTypeServiceImpl.deleteWidgetTypesByTenantIdAndBundleAlias(WidgetTypeServiceImpl.java:93)
	at org.thingsboard.server.dao.widget.WidgetsBundleServiceImpl.deleteWidgetsBundle(WidgetsBundleServiceImpl.java:78)
	at org.thingsboard.server.service.install.DefaultSystemDataLoaderService.deleteSystemWidgetBundle(DefaultSystemDataLoaderService.java:161)
	at org.thingsboard.server.install.ThingsboardInstallService.performInstall(ThingsboardInstallService.java:129)
	at org.thingsboard.server.ThingsboardInstallApplication.main(ThingsboardInstallApplication.java:43)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:593)
Unexpected error during ThingsBoard installation!
waiting for server to shut down............................................................... failed
pg_ctl: server does not shut down
```

**Comment by stefanfoulis** [comment]
looks like my db is not _that_ small after all:
```
root@84c027c57ca6:/data/db# du -hd1
4.0K	./pg_twophase
28K	./pg_multixact
12K	./pg_logical
4.0K	./pg_stat
4.0K	./pg_serial
4.0K	./pg_commit_ts
43M	./pg_clog
508K	./global
4.0K	./pg_snapshots
12K	./pg_notify
256K	./pg_subtrans
81M	./pg_xlog
4.0K	./pg_tblspc
4.0K	./pg_dynshmem
48G	./base
4.0K	./pg_replslot
32K	./pg_stat_tmp
48G	.
```
I'll run a `vacuum` on postgres and retry running the upgrade script. Maybe that helps.

**Comment by stefanfoulis** [comment]
🤔 maybe I should switch to cassandra instead #591?

**Comment by stefanfoulis** [comment]
`vacuum` has been running for a long time... this may actually be the issue. Let's see what happens once it finishes 🤞.

**Comment by ban2derlog** [comment]
Hi, do you resolved issue with update procedure to 2.4.2 in Docker?
